pageserver/src/deletion_queue/backend.rs

```diff
@@ -0,0 +1,343 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::Duration;
+
+use tokio_util::sync::CancellationToken;
+use tracing::debug;
+use tracing::info;
+use tracing::warn;
+
+use crate::config::PageServerConf;
+use crate::control_plane_client::ControlPlaneGenerationsApi;
+use crate::metrics::DELETION_QUEUE_DROPPED;
+use crate::metrics::DELETION_QUEUE_ERRORS;
+
+use super::executor::ExecutorMessage;
+use super::DeletionHeader;
+use super::DeletionList;
+use super::DeletionQueueError;
+use super::FlushOp;
+use super::VisibleLsnUpdates;
+
+// After this length of time, do any validation work that is pending,
+// even if we haven't accumulated many keys to delete.
+//
+// This also causes updates to remote_consistent_lsn to be validated, even
+// if there were no deletions enqueued.
+const AUTOFLUSH_INTERVAL: Duration = Duration::from_secs(10);
+
+// If we have received this number of keys, proceed with attempting to execute
+const AUTOFLUSH_KEY_COUNT: usize = 16384;
+
+#[derive(Debug)]
+pub(super) enum BackendQueueMessage {
+    Delete(DeletionList),
+    Flush(FlushOp),
+}
+pub struct BackendQueueWorker {
```

Comment:
What is the point of backend/frontend split? Could describe in type comment.

It's a bit unfortunate that this needs to be pub.

Reply:
Ah and there's also the executor :)

Reply:
This was refactored to no longer be pub in b76f08e8634256ad8d2000b69e49e18f26d0fcbc.

The frontend/backend/executor separation is explained in the doc comment for DeletionQueue.  As with most things, we _could_ squish them together into one type, but their functionality is kinda separate:
- The frontend collects layer deletions into DeletionLists and saves them to disk
- The backend validates deletionlists and remote consistent LSN updates
- The executor coalesces deletions into large DeleteObjects requests.